src/Flow.cpp

```diff
@@ -1700,16 +1700,28 @@ json_object* Flow::flow2json() {
 
   if((my_object = json_object_new_object()) == NULL) return(NULL);
 
-  if(ntop->getPrefs()->do_dump_flows_on_es()) {
+  if(ntop->getPrefs()->do_dump_flows_on_es()
+    || ntop->getPrefs()->do_dump_flows_on_ls()
+    ) {
     struct tm* tm_info;
 
     t = last_seen;
     tm_info = gmtime(&t);
 
     strftime(buf, sizeof(buf), "%FT%T.0Z", tm_info);
-    json_object_object_add(my_object, "@timestamp", json_object_new_string(buf));
+
+    if(ntop->getPrefs()->do_dump_flows_on_ls()){
+      /*  Add current timestamp differently for Logstash, in case of delay 
+       *  Note: Logstash generates it's own @timestamp field on input
+       */
+      json_object_object_add(my_object,"ntop_timestamp",json_object_new_string(buf));
```

Comment:
There will be a delay between the actual insert into Elasticsearch by default so the timestamp will be different. If we send the original timestamp and replace the Logstash generated value of @timestamp the data will be mapped better in visualizers. 
 We can delete the ntop_timestamp field afterwards in logstash so we don't have an extra field.